src/util/system.cpp

```diff
@@ -388,6 +388,45 @@ std::optional<unsigned int> ArgsManager::GetArgFlags(const std::string& name) co
     return std::nullopt;
 }
 
+const fs::path& ArgsManager::GetDataDirPath(bool fNetSpecific)
+{
+    LOCK(cs_args);
+    fs::path &path = fNetSpecific ? datadirPathCachedNetSpecific : datadirPathCached;
+
+    // Cache the path to avoid calling fs::create_directories on every call of
+    // this function
+    if (!path.empty()) return path;
+
+    std::string datadir = GetArg("-datadir", "");
+    if (!datadir.empty()) {
+        path = fs::system_complete(datadir);
+        if (!fs::is_directory(path)) {
+            path = "";
+            return path;
+        }
+    } else {
+        path = GetDefaultDataDir();
+    }
+    if (fNetSpecific)
+        path /= BaseParams().DataDir();
+
+    if (fs::create_directories(path)) {
+        // This is the first run, create wallets subdirectory too
+        fs::create_directories(path / "wallets");
+    }
+
+    path = StripRedundantLastElementsOfPath(path);
+    return path;
+}
+
+void ArgsManager::ClearDatadirPathCache()
```

Comment:
In commit "Move GetDataDir(fNetSpecific) implementation to ArgsManager." (839324993aeb8448af5a811b7323fa460ac9eac7)

This commit seems not to compile

```
util/system.cpp:422:19: error: out-of-line definition of 'ClearDatadirPathCache' does not match any declaration in 'ArgsManager'; did you mean 'ClearPathCache'?
void ArgsManager::ClearDatadirPathCache()
                  ^~~~~~~~~~~~~~~~~~~~~
                  ClearPathCache                                                                                                                                          
./util/system.h:280:10: note: 'ClearPathCache' declared here
    void ClearPathCache();
         ^
util/system.cpp:820:11: error: no member named 'ClearDatadirPathCache' in 'ArgsManager'
    gArgs.ClearDatadirPathCache();
    ~~~~~ ^
2 errors generated.
```

Reply:
This is fixed on my system now. Thanks.